### DISCUSSION TITLE
Handing a u128 to Java.

### DISCUSSION THREAD

**Discussion by IoIxD** [jni-rs/jni-rs#464] [General]
Java has a type called "java.util.UUID" which, despite its fancy methods, is functionally equivalent Rust's `u128` type. So I was hoping this would work:
```
pub fn entity(jni: &mut jni::JNIEnv<'mc>,arg0: u128)  {
    let val_0 = jni.new_object("java/util/UUID", "()V", &[arg0.into()]);
}
```
It does not.
This would obviously be worth opening a feature request for but I was curious if there was some sort of hack or alternative code I could try?

**Comment by argv-minus-one** [comment]
`java.util.UUID` is actually two `i64`s in a trenchcoat. This function should do the trick:

```rust
fn uuid_to_java<'local>(
	env: &mut JNIEnv<'local>,
	uuid: u128,
) -> jni::errors::Result<JObject<'local>> {
	let upper = (uuid >> 64) as u64 as i64;
	let lower = uuid as u64 as i64;

	env.new_object("java/util/UUID", "(JJ)V", &[upper.into(), lower.into()])
}
```

**Comment by IoIxD** [comment]
I was getting ready to use fromString but this is even better. Thanks!
